pkg/sippyserver/server.go

```diff
@@ -363,22 +358,36 @@ func (s *Server) variantsReport(w http.ResponseWriter, req *http.Request) {
 }
 
 func (s *Server) Serve() {
-	http.DefaultServeMux.HandleFunc("/", s.printHTMLReport)
-	http.DefaultServeMux.HandleFunc("/install", s.printInstallHTMLReport)
-	http.DefaultServeMux.HandleFunc("/upgrade", s.printUpgradeHTMLReport)
-	http.DefaultServeMux.HandleFunc("/operator-health", s.printOperatorHealthHTMLReport)
-	http.DefaultServeMux.HandleFunc("/testdetails", s.printTestDetailHTMLReport)
-	http.DefaultServeMux.HandleFunc("/json", s.printJSONReport)
-	http.DefaultServeMux.HandleFunc("/detailed", s.detailed)
-	http.DefaultServeMux.HandleFunc("/refresh", s.refresh)
-	http.DefaultServeMux.HandleFunc("/canary", s.printCanaryReport)
-	http.DefaultServeMux.HandleFunc("/api/jobs", s.jobs)
-	http.DefaultServeMux.HandleFunc("/jobs", s.jobsReport)
-	http.DefaultServeMux.HandleFunc("/variants", s.variantsReport)
-
-	http.DefaultServeMux.Handle("/static/", http.StripPrefix("/static/", http.FileServer(http.Dir("./static"))))
+	// Use private ServeMux to prevent tests from stomping on http.DefaultServeMux
+	serveMux := http.NewServeMux()
+
+	serveMux.HandleFunc("/", s.printHTMLReport)
+	serveMux.HandleFunc("/install", s.printInstallHTMLReport)
+	serveMux.HandleFunc("/upgrade", s.printUpgradeHTMLReport)
+	serveMux.HandleFunc("/operator-health", s.printOperatorHealthHTMLReport)
+	serveMux.HandleFunc("/testdetails", s.printTestDetailHTMLReport)
+	serveMux.HandleFunc("/json", s.printJSONReport)
+	serveMux.HandleFunc("/detailed", s.detailed)
+	serveMux.HandleFunc("/refresh", s.refresh)
+	serveMux.HandleFunc("/canary", s.printCanaryReport)
+	serveMux.HandleFunc("/api/jobs", s.jobs)
+	serveMux.HandleFunc("/jobs", s.jobsReport)
+	serveMux.HandleFunc("/variants", s.variantsReport)
+	serveMux.Handle("/static/", http.StripPrefix("/static/", http.FileServer(http.Dir("./static"))))
```

Comment:
don't you still need to add the defaulthandler?  (for 404s if nothing else)

Reply:
You mean https://github.com/openshift/sippy/pull/252/files#diff-768a5e234f25507b29e6156fd5af2bfb561aef7c7b19612e7e73ea4cb6be07e4L93-L99? It doesn't appear to be wired into anything.

Instead, it looks like `server.printHTMLReport` is what's wired up to the root handler (https://github.com/openshift/sippy/pull/252/files#diff-768a5e234f25507b29e6156fd5af2bfb561aef7c7b19612e7e73ea4cb6be07e4L93-L99) and what handles 404s (which it does by silently redirecting to the landing page) (https://github.com/openshift/sippy/blob/master/pkg/sippyserver/server.go#L101-L107).

Reply:
hm, there's some weirdness here but yeah i guess that other code is dead.